tests/src/services/HeadersTests.scala

```diff
@@ -0,0 +1,289 @@
+import spray.http.AllOrigins
+import spray.http.BasicHttpCredentials
+import spray.http.HttpHeader
+import spray.http.HttpHeaders.`Access-Control-Allow-Headers`
+import spray.http.HttpHeaders.`Access-Control-Allow-Origin`
+import spray.http.HttpMethod
+import spray.http.HttpMethods.CONNECT
+import spray.http.HttpMethods.DELETE
+import spray.http.HttpMethods.GET
+import spray.http.HttpMethods.HEAD
+import spray.http.HttpMethods.OPTIONS
+import spray.http.HttpMethods.PATCH
+import spray.http.HttpMethods.POST
+import spray.http.HttpMethods.PUT
+import spray.http.HttpMethods.TRACE
+import spray.http.HttpRequest
+import spray.http.HttpResponse
+import spray.http.StatusCodes.Accepted
+import spray.http.StatusCodes.OK
+import spray.http.Uri
+import spray.http.Uri.Path
+import spray.testkit.ScalatestRouteTest
+
+@RunWith(classOf[JUnitRunner])
+class HeadersTests extends FlatSpec
+    with ScalatestRouteTest
+    with Matchers {
+
+    behavior of "Headers at general API"
+
+    val timeout = 60 seconds
+
+    val whiskAuth = WhiskProperties.getBasicAuth
+    val creds = BasicHttpCredentials(whiskAuth.fst, whiskAuth.snd)
+
+    val allMethods = Set(DELETE, GET, POST, PUT)
+    val allowOrigin = `Access-Control-Allow-Origin`(AllOrigins)
+    val allowHeaders = `Access-Control-Allow-Headers`("Authorization", "Content-Type")
+
+    val url = Uri(s"http://${WhiskProperties.getControllerHost}:${WhiskProperties.getControllerPort}")
+    val pipeline: HttpRequest => Future[HttpResponse] = (
+        sendReceive
+        ~> unmarshal[HttpResponse])
+
+    val basePath = Path("/api/v1")
+
+    def toHttpMethod(method: String): Option[HttpMethod] = {
+        method.trim match {
+            case "CONNECT" => Some(CONNECT)
+            case "DELETE"  => Some(DELETE)
+            case "GET"     => Some(GET)
+            case "HEAD"    => Some(HEAD)
+            case "OPTIONS" => Some(OPTIONS)
+            case "PATCH"   => Some(PATCH)
+            case "POST"    => Some(POST)
+            case "PUT"     => Some(PUT)
+            case "TRACE"   => Some(TRACE)
+            case x         => None
+        }
+    }
+
+    /**
+     * Checks, if the required headers are in the list of all headers.
+     * For the allowed method, it checks, if only the allowed methods are in the response headers.
+     */
+    def containsHeaders(headers: List[HttpHeader], allowedMethods: Set[HttpMethod]) = {
+        val headersMap = headers.map { header =>
+            header.name -> header.value.split(",").flatMap(toHttpMethod(_)).toSet
+        } toMap
+
+        headers should contain allOf (allowOrigin, allowHeaders)
+        //        headersMap should contain key "Access-Control-Allow-Methods"
+        //
+        //        headersMap("Access-Control-Allow-Methods") should contain theSameElementsAs (allowedMethods)
+    }
+
+    it should "respond to OPTIONS with all headers" in {
+        val response = Await.result(pipeline(Options(url.withPath(basePath))), timeout)
```

Comment:
Can we use http://www.artima.com/docs-scalatest-2.0.M5/org/scalatest/concurrent/Futures.html to have more readable tests?